plugins/module_utils/network/sonic/config/vrrp/vrrp.py

```diff
@@ -0,0 +1,747 @@
+        is_delete_all = False
+        if not want:
+            commands = self.get_all_vrrps(have)
+            is_delete_all = True
+        else:
+            self.sort_lists_in_config(want)
+            self.sort_lists_in_config(have)
+            new_want = deepcopy(want)
+            for default_entry in default_entries:
+                remove_matching_defaults(new_want, default_entry)
+            new_want = remove_empties_from_list(new_want)
+            commands = new_want
+
+        requests = self.get_delete_vrrp_requests(commands, have, is_delete_all)
+
+        if commands and len(requests) > 0:
+            commands = update_states(commands, 'deleted')
+        else:
+            commands = []
+
+        return commands, requests
+
+    def _get_replaced_overridden_config(self, want, have, state):
+        add_config, del_config = [], []
+        for cmd in want:
+            name = cmd.get('name')
+            groups = cmd.get('group')
+            match = next((m for m in have if m['name'] == name), None)
+
+            if not match:
+                add_config.append(cmd)
+            else:
+                add_cfg, del_cfg = {}, {}
+                for group in groups:
+                    vr_id = group.get('virtual_router_id')
+                    afi = group.get('afi')
+                    match_group = next((g for g in match['group'] if g['virtual_router_id'] == vr_id and g['afi'] == afi), None)
+                    if not match_group:
+                        add_cfg.setdefault('group', []).append(group)
+                    else:
+                        add_group, del_group = {}, {}
+                        for attr in self.vrrp_attributes:
+                            if attr in group:
+                                if attr not in match_group:
+                                    add_group[attr] = group[attr]
+                                elif group[attr] != match_group[attr]:
+                                    if attr not in ('virtual_address', 'track_interface'):
+                                        add_group[attr] = group[attr]
+                                        del_group[attr] = match_group[attr]
+                                    elif attr == 'virtual_address':
+                                        add_vip, del_vip = [], []
+                                        for vip in group[attr]:
+                                            match_vip = next((v for v in match_group[attr] if v['address'] == vip['address']), None)
+                                            if not match_vip:
+                                                add_vip.append(vip)
+                                        for match_vip in match_group[attr]:
+                                            vip = next((v for v in group[attr] if v['address'] == match_vip['address']), None)
+                                            if not vip:
+                                                del_vip.append(match_vip)
```

Comment:
Could be simplified by using sets for obtaining the difference between have and want (get_vip_addresses method can be used to get the elements)